frappe/client.py

```diff
@@ -430,6 +430,16 @@ def validate_link(doctype: str, docname: str, fields=None):
 			frappe.PermissionError,
 		)
 
+	if frappe.get_meta(doctype).is_virtual:
```

Comment:
Use `is_virtual_doctype` from model/utils for better perf here. We don't want to get meta for every doctype. (most common case would be non-virtual doctypes)